### PR TITLE
Fix GET /agents/stats/distinct

### LINKED ISSUE #8538: Investigate and fix error when including id in fields parameter in GET /agents/stats/distinct endpoint
|Wazuh branch|Component|
|---|---|
| master | API/Framework |

Hi team,

While testing the  `GET /agents/stats/distinct endpoint`, I have discovered a wrong behaviour.

If we include `id` in the `fields` parameter, the endpoint returns what is expected:

#### Request:

`/agents/stats/distinct?fields=node_name,id`

```
{
  "data": {
    "affected_items": [
      {
        "id": "000",
        "node_name": "master-node",
        "count": 1
      },
      {
        "id": "001",
        "node_name": "master-node",
        "count": 1
      },
      {
        "id": "002",
        "node_name": "master-node",
        "count": 1
      }
    ],
    "total_affected_items": 3,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected agents information was returned",
  "error": 0
}
```

If we include `id` in the `fields` parameter, and a `select` without `id` the endpoint returns all `id` values as `unknown`. This also happens when we leave `fields` empty (it also takes `id`).

#### Request:

`/agents/stats/distinct?fields=node_name,id&select=node_name
`

```
{
  "data": {
    "affected_items": [
      {
        "node_name": "master-node",
        "count": 1,
        "id": "unknown"
      },
      {
        "node_name": "master-node",
        "count": 1,
        "id": "unknown"
      },
      {
        "node_name": "master-node",
        "count": 1,
        "id": "unknown"
      }
    ],
    "total_affected_items": 3,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected agents information was returned",
  "error": 0
}
```

#### Request:

`/agents/stats/distinct?select=node_name
`

```
{
  "data": {
    "affected_items": [
      {
        "node_name": "master-node",
        "count": 1,
        "id": "unknown"
      },
      {
        "node_name": "master-node",
        "count": 1,
        "id": "unknown"
      },
      {
        "node_name": "master-node",
        "count": 1,
        "id": "unknown"
      }
    ],
    "total_affected_items": 3,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected agents information was returned",
  "error": 0
}
```

Investigate and fix this error, the normal behavior must be not returning id if we haven't selected it.

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### PR DESCRIPTION
**Pull request by Selutario**
|Related issue|
|---|
| Closes #8538 |

## Description

Hi team!

This PR fixes the error in `GET /agents/stats/distinct` that was reported in #8538. Now, ID field is not included unless the user specifies it, so it will not be shown as `unknown`:

> GET /agents/stats/distinct?fields=node_name,id&select=node_name

```JSON
{
  "data": {
    "affected_items": [
      {
        "node_name": "master-node",
        "count": 1
      },
      {
        "node_name": "unknown",
        "count": 1
      },
      {
        "node_name": "unknown",
        "count": 1
      },
      {
        "node_name": "worker1",
        "count": 1
      },
      {
        "node_name": "unknown",
        "count": 1
      },
      {
        "node_name": "unknown",
        "count": 1
      }
    ],
    "total_affected_items": 6,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All selected agents information was returned",
  "error": 0
}
```

In addition, a bug has been fixed. Before, if `fields` and `select` used different values, an error like the one below was raised:

> GET /agents/stats/distinct?fields=os.codename&select=node_name
```JSON
{
  "title": "Bad Request",
  "detail": "Error in wazuhdb request: Cannot execute Global database query; near \"from\": syntax error",
  "remediation": "Make sure the request is correct",
  "dapi_errors": {
    "master-node": {
      "error": "Error in wazuhdb request: Cannot execute Global database query; near \"from\": syntax error"
    }
  },
  "error": 2003
}
```

Now, an exception with a more explanatory message has been added:
```JSON
{
  "title": "Bad Request",
  "detail": "Select fields have to be a subset of the filter fields: select fields: ['node_name'], filter fields: ['os.codename']",
  "dapi_errors": {
    "master-node": {
      "error": "Select fields have to be a subset of the filter fields: select fields: ['node_name'], filter fields: ['os.codename']"
    }
  },
  "error": 1760
}
```

Regards,
Selu.

### CODE REVIEW

File: api/test/integration/test_agent_GET_endpoints.tavern.yaml [hunk 1]
```diff
@@ -3846,13 +3846,28 @@ stages:
...
       params:
         limit: 2
-        fields: os.platform
         select: invalid
     response:
       status_code: 400
       json:
         error: 1724
 
+  - name: Select is different to filter fields
```
Comment (CarlosRS9) [inline api/test/integration/test_agent_GET_endpoints.tavern.yaml:3855]:
```suggestion
  - name: Select is different from filter fields
```

**Review by CarlosRS9 — CHANGES_REQUESTED** [review]

**Review by CarlosRS9 — APPROVED** [review]

**Comment by mcarmona99** [comment]
### PR update

We have decided to remove the select parameter from the `GET /agents/stats/distinct` endpoint.

With this change, the error will be fixed. `fields` will be the parameter to indicate how to distinct and select.

- Test results:

```
test_agent_GET_endpoints.tavern.yaml 
	 91 passed, 93 warnings
```

```
framework/wazuh/core/tests/
======= 667 passed, 1 xfailed, 2 xpassed, 9 warnings in 4.21s =======

framework/wazuh/tests/
======= 587 passed, 13 warnings in 54.29s =======
```

Regards,
Manuel.

**Review by vicferpoy — APPROVED** [review]
LGTM!

**Review by vicferpoy — APPROVED** [review]
LGTM!

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**